pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2494,50 +2494,71 @@ public void readEntryComplete(Entry entry, Object ctx) {
         }
     }
 
-    protected Response internalPeekNthMessage(String subName, int messagePosition, boolean authoritative) {
+    protected void internalPeekNthMessage(AsyncResponse asyncResponse, String subName, int messagePosition,
+                                              boolean authoritative) {
         // If the topic name is a partition name, no need to get partition topic metadata again
         if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName,
```

Comment:
getPartitionedTopicMetadata looks like a sync method.

Reply:
Thank you. Done